ceph.spec.in

```diff
@@ -797,18 +800,42 @@ export CPPFLAGS="$java_inc"
 export CFLAGS="$RPM_OPT_FLAGS"
 export CXXFLAGS="$RPM_OPT_FLAGS"
 
-env | sort
-
+# Parallel build settings ...
+# unlimit _smp_mflags in system macro
+%global _smp_ncpus_max 0
+CEPH_PARALLEL_BUILD="%{?_smp_mflags}"
 %if %{with lowmem_builder}
-%if 0%{?jobs} > 8
-%define _smp_mflags -j8
-%endif
+lo_jobs="%{?jobs:%{jobs}}"
+%if 0%{?suse_version}
+%ifarch %arm
+    %if ! 0%{?qemu_user_space_build:1}
+        # native hardware is not that powerful
+        lo_jobs="1"
+    %endif
+%endif
+%endif
+# do not eat all memory
+echo "Available memory:"
+free
+echo "System limits:"
+ulimit -a
+lo_jobs_reduced=
+if test -n "$lo_jobs" -a "$lo_jobs" -gt 1 ; then
+    mem_per_process=1600
+    max_mem=$(LANG=C free -t -m | sed -n "s|^Mem: *\([0-9]*\).*$|\1|p")
+    max_jobs="$(($max_mem / $mem_per_process))"
+    test "$lo_jobs" -gt "$max_jobs" && lo_jobs="$max_jobs" && lo_jobs_reduced="yes" && echo "Warning: Reducing build parallelism to -j$max_jobs because of memory limits"
+    test "$lo_jobs" -le 0 && lo_jobs=1 && echo "Warning: Not using parallel build at all because of memory limits"
+fi
+# run make
+if test -n "$lo_jobs" ; then
+  CEPH_PARALLEL_BUILD="-j$lo_jobs"
+fi
 %endif
-
-# unlimit _smp_mflags in system macro if not set above
-%define _smp_ncpus_max 0
-# extract the number of processors for use with cmake
-%define _smp_ncpus %(echo %{_smp_mflags} | sed 's/-j//')
+export CEPH_PARALLEL_BUILD
+export CEPH_SMP_NCPUS=$(echo "$CEPH_PARALLEL_BUILD" | sed 's/-j//')
```

Comment:
no need to `export` these two variables. we *pass* them to `cmake` and `make` explicitly.

Reply:
I'm exporting them so they appear in `env` so the user can see their values in the build log.

Reply:
Do you think the export is dangerous in this case? If not, I would argue that it's beneficial to see the values in the build log...